src/Components/ADT3DSceneBuilder/Internal/VisualRuleForm/Internal/ConditionsCallout/ConditionsCallout.tsx

```diff
@@ -0,0 +1,255 @@
+import React, { useCallback, useEffect, useReducer, useRef } from 'react';
+import {
+    ActionItemKey,
+    ConditionCalloutActionType,
+    ConditionCalloutReducerType,
+    ConditionValidityMap,
+    IConditionsCalloutProps,
+    IConditionsCalloutStyleProps,
+    IConditionsCalloutStyles
+} from './ConditionsCallout.types';
+import { getStyles } from './ConditionsCallout.styles';
+import {
+    classNamesFunction,
+    useTheme,
+    styled,
+    Callout,
+    Stack,
+    DirectionalHint,
+    TextField,
+    PrimaryButton,
+    DefaultButton
+} from '@fluentui/react';
+import { ConditionSummary } from './ConditionSummary';
+import { ActionItem } from './ActionItem';
+import { ConditionCalloutReducer } from './ConditionsCallout.state';
+import { IDTDLPropertyType } from '../../../../../../Models/Types/Generated/3DScenesConfiguration-v1.0.0';
+import { useTranslation } from 'react-i18next';
+import { areRangesValid, checkValidity } from './ConditionCalloutUtility';
+import { deepCopy } from '../../../../../../Models/Services/Utils';
+import { CalloutInfoType } from '../ConditionsList.types';
+
+const getClassNames = classNamesFunction<
+    IConditionsCalloutStyleProps,
+    IConditionsCalloutStyles
+>();
+
+const ROOT_LOC = '3dSceneBuilder.visualRuleForm';
+const LOC_KEYS = {
+    addCondition: `${ROOT_LOC}.addCondition`,
+    editCondition: `${ROOT_LOC}.editCondition`,
+    labelField: `${ROOT_LOC}.labelField`
+};
+
+const ConditionsCallout: React.FC<IConditionsCalloutProps> = (props) => {
+    const {
+        calloutType,
+        isOpen,
+        onDismiss,
+        onSave,
+        target,
+        styles,
+        valueRange,
+        valueRangeType
+    } = props;
+
+    // state
+
+    // refs
+    const validityMap = useRef<ConditionValidityMap>({
+        label: !!valueRange.visual.labelExpression?.length,
+        ranges: areRangesValid(valueRange.values, valueRangeType)
+    });
+
+    // hooks
+    const { t } = useTranslation();
+    const [
+        conditionCalloutState,
+        conditionCalloutDispatch
+    ] = useReducer<ConditionCalloutReducerType>(ConditionCalloutReducer, {
+        conditionToEdit: valueRange,
+        originalCondition: valueRange,
+        isDirty: false
+    });
+
+    // side-effects
+    useEffect(() => {
+        // RESET CONDITION IN STATE AND VALIDITY MAP
+        conditionCalloutDispatch({
+            type: ConditionCalloutActionType.RESET_CONDITION,
+            payload: { valueRange: valueRange }
+        });
+        validityMap.current = {
+            label: !!valueRange.visual.labelExpression?.length,
+            ranges: areRangesValid(valueRange.values, valueRangeType)
+        };
+    }, [valueRange, valueRangeType]);
+
+    // callbacks
+    const onLabelChange = useCallback(
+        (
+            _event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>,
+            newValue?: string
+        ) => {
+            if (newValue) {
```

Comment:
nit: might want to trim the value here just to avoid setting to valid when it's empty string.